### ISSUE TITLE
UI should be live with information pushed from server

### ISSUE THREAD

**Issue by andrewrk** [kubernetes/dashboard#2541]
There's no reason to press a refresh button. Instead of the dashboard requesting information from the server, it should subscribe to information from the server, and update the UI to reflect changes from the server.

**Comment by maciaszczykm** [comment]
Agreed. Auto-refresh would be a temporary solution in this case as it is easier to implement at the moment.

**Comment by hexsprite** [comment]
You could look at using GraphQL as it supports real-time data Subscriptions and has great cross platform support.

**Comment by javajon** [comment]
For inspiration [CoreOS Tectonic](https://coreos.com/tectonic/sandbox/) dashboard for Kubernetes updates dynamically.

**Comment by Antiarchitect** [comment]
Just tracking this.

**Comment by kachkaev** [comment]
As a step towards this, would it be reasonable to implemented a refresh button to re-fetch lists of pods, deployments, etc? I know one can just refresh the entire page, but that's not quite the same. List refresh can work much faster because it won't require css and js reloading, which is especially noticeable on slow remote networks.

I imagine it can look like this:

![refresh-list](https://user-images.githubusercontent.com/608862/36629789-f211e5d4-1952-11e8-8e9a-1e7b62433307.png)


Or like this:

![refresh-all](https://user-images.githubusercontent.com/608862/36629793-f6d8b2a0-1952-11e8-8212-f32b879d231f.png)

**Comment by floreks** [comment]
We are considering few solutions.

1. Adding auto-refresh option to local settings so user can select i.e. that he wants data to be refreshed every x sec.
2. Thanks to angular migration it will be possible to support watch for resources as apiserver exposes watch endpoints. 

We will revisit this after we finish angular migration.

**Comment by kachkaev** [comment]
Good to know! If you consider adding auto-refresh option in future, it'd be also useful to have a tick top stops data refreshing when the tab is not focused. Here is how this setting looks like in [netdata](https://github.com/firehol/netdata) (server monitoring tool):

<img width="556" alt="screen shot 2018-02-24 at 11 58 57" src="https://user-images.githubusercontent.com/608862/36630196-57da0624-195a-11e8-9d31-a5d81bdc7462.png">

**Comment by maciaszczykm** [comment]
@nebrass It is frozen because otherwise, bot would automatically close it after 30 days. I agree that it is very important, but even more important for us was the update to the new Angular as some of the old stuff was going to be deprecated and we wanted to use new functionality. Unfortunately, at the moment I have no time to contribute anymore so let's hope for contributions from other people to push migration and this feature forward.

**Comment by spacecat** [comment]
13 apr. Last post. Long time ago. Just checking on status? Also, my first though was "Click on a category in the left nav and the right / main view should reload/refresh. Now I have to click back and forth between categories in the left nav to update the right view. Dynamically/auto-update by subscribing to the server definitely would be the ultimate solution as already mentioned.

**Comment by benc-uk** [comment]
I would love to see this, I run a lot of demos of Kubernetes and having the dashboard auto refresh would make my life so much easier. It would make the dashboard feel a lot more modern and responsive

**Comment by zedtux** [comment]
I'm also looking for this feature that would make the dashboard so much better! Please ! 🙏

**Comment by techfort** [comment]
any update on this?

**Comment by floreks** [comment]
There is a simple resource auto refresh added to the latest beta. It's not based on websockets thought.
